spec/integration/federation/federation_messages_generation.rb

```diff
@@ -0,0 +1,178 @@
+      entity,
+      OpenSSL::PKey::RSA.new(user.encryption_key)
+    )
+  end
+end
+
+def generate_profile
+  @entity = FactoryGirl.build(:profile_entity, diaspora_id: @remote_person.diaspora_handle)
+
+  generate_xml(@entity, @remote_user, @user)
+end
+
+def generate_conversation
+  @entity = FactoryGirl.build(
+    :conversation_entity,
+    diaspora_id:     @remote_user.diaspora_handle,
+    participant_ids: "#{@remote_user.diaspora_handle};#{@user.diaspora_handle}"
+  )
+
+  generate_xml(@entity, @remote_user, @user)
+end
+
+def generate_status_message
+  @entity = FactoryGirl.build(
+    :status_message_entity,
+    diaspora_id: @remote_user.diaspora_handle,
+    public:      @public
+  )
+
+  generate_xml(@entity, @remote_user, @user)
+end
+
+def generate_forged_status_message
+  substitute_wrong_key(@remote_user, 1)
+  generate_status_message
+end
+
+def generate_reshare
+  @entity = FactoryGirl.build(
+    :reshare_entity,
+    root_diaspora_id: @user.diaspora_handle,
+    root_guid:        @local_target.guid,
+    diaspora_id:      @remote_user.diaspora_handle,
+    public:           true
+  )
+
+  generate_xml(@entity, @remote_user, @user)
+end
+
+def mock_private_key_for_user(user)
+  expect(DiasporaFederation.callbacks).to receive(:trigger)
+                                            .with(:fetch_private_key_by_diaspora_id, user.person.diaspora_handle)
+                                            .once
+                                            .and_return(user.encryption_key)
+end
+
+def retraction_mock_callbacks(entity, sender)
+  return unless [
+    DiasporaFederation::Entities::SignedRetraction,
+    DiasporaFederation::Entities::RelayableRetraction
+  ].include?(entity.class)
+
+  mock_private_key_for_user(sender)
+
+  allow(DiasporaFederation.callbacks).to receive(:trigger)
+                                            .with(
+                                              :fetch_entity_author_id_by_guid,
+                                              entity.target_type,
+                                              entity.target_guid
+                                            )
+                                            .once
+                                            .and_return(sender.encryption_key)
+end
+
+def generate_retraction(entity_name, target_object, sender=@remote_user)
+  @entity = FactoryGirl.build(
+    entity_name,
+    diaspora_id: sender.diaspora_handle,
+    target_guid: target_object.guid,
+    target_type: target_object.class.to_s
+  )
+
+  retraction_mock_callbacks(@entity, sender)
+
+  generate_xml(@entity, sender, @user)
+end
+
+def generate_forged_retraction(entity_name, target_object, sender=@remote_user)
+  times = 1
+  if %i(signed_retraction_entity relayable_retraction_entity).include?(entity_name)
+    times += 2
+  end
+
+  substitute_wrong_key(sender, times)
+  generate_retraction(entity_name, target_object, sender)
+end
+
+def generate_relayable_local_parent(entity_name)
```

Comment:
Assignment Branch Condition size for generate_relayable_local_parent is too high. [22.11/20]